flink-connectors/flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSGeneralUtil.java

```diff
@@ -87,6 +88,20 @@ public static CredentialProvider getCredentialProviderType(
         }
     }
 
+    /**
+     * Return a {@link AwsCredentialsProvider} instance corresponding to the configuration
+     * properties.
+     *
+     * @param configProps the configuration property map
+     * @return The corresponding AWS Credentials Provider instance
+     */
+    public static AwsCredentialsProvider getCredentialsProvider(final Map<String, ?> configProps) {
+        Properties properties = new Properties();
+        properties.putAll(configProps);
+
+        return getCredentialsProvider(properties, AWSConfigConstants.AWS_CREDENTIALS_PROVIDER);
```

Comment:
nit: would `return getCredentialsProvider(properties);` be nicer for this line (using the helper on line 112)